src/zarr/buffer.py

```diff
@@ -464,9 +417,32 @@ def as_numpy_array_wrapper(func: Callable[[npt.NDArray[Any]], bytes], buf: Buffe
     buf
         The buffer that will be converted to a Numpy array before given as
         input to `func`.
+    prototype
+        The prototype of the output buffer.
+
+    Returns
+    -------
+        The result of `func` converted to a `prototype.buffer`
+    """
+    return prototype.buffer.from_bytes(func(buf.as_numpy_array()))
+
+
+class Prototype(NamedTuple):
```

Comment:
I am unsure whether the name `Prototype` conveys what the purpose of this class is. Is that a common way of calling such a construct? Maybe we could make it less general: `BufferPrototype`. @d-v-b any opinion on this?

Reply:
I think `Template` would be a better name.

Reply:
i think prototype works, but I think the need for this class suggests that we might want to re-consider the hard boundary we have drawn between buffers and ndbuffers.

Reply:
I have no strong opinion, `Template` or  `BufferPrototype` are also good names or maybe `BufferTemplate` ?

Reply:
No strong opinion, either, but if we had to decide, I'd go with `BufferPrototype`.

Reply:
Renamed to `BufferPrototype`